src/pages/SubmissionDetails.js

```diff
@@ -179,7 +166,7 @@ export const SubmissionDetails = ({ submissionId }) => {
         h2(['Submission name: ', specifyRunSet[0]?.run_set_name]),
         h3(['Workflow name: ', getSpecificMethod[0]?.name]),
         h3(['Submission date: ', specifyRunSet[0] && makeCompleteDate(specifyRunSet[0].submission_timestamp)]),
-        h3(['Duration: ', specifyRunSet[0] && customFormatDuration(duration(specifyRunSet[0]))])
+        h3(['Duration: ', specifyRunSet[0] && customFormatDuration(getDuration(specifyRunSet[0].state, specifyRunSet[0].submission_timestamp, specifyRunSet[0].last_modified_timestamp, isRunSetInTerminalState))])
```

Comment:
This is the fix for the bug listed in the ticket. We were incorrectly using `isRunInTerminalState` to check if Run Set was in terminal state. 

Other changes are to reduce code redundancy around getting duration value and making it more easily testable.